skins/laika/tests/unit/components/pages/donation_form/BankData.spec.ts

```diff
@@ -0,0 +1,173 @@
+		const wrapper = mount( BankData, {
+			localVue,
+			propsData: {
+				validateBankDataUrl: '/check-iban',
+				validateLegacyBankDataUrl: '/generate-iban',
+			},
+			store: createStore(),
+			mocks: {
+				$t: () => {},
+			},
+		} );
+		const store = wrapper.vm.$store;
+		store.dispatch = jest.fn();
+
+		const iban = wrapper.find( '#iban' );
+		wrapper.setData( { accountId: 'DE12345605171238489890' } );
+		iban.trigger( 'blur' );
+		const expectedAction = action( NS_BANKDATA, setBankData );
+		const expectedPayload = {
+			validationUrl: '/check-iban',
+			requestParams: { iban: 'DE12345605171238489890' },
+		} as BankAccountRequest;
+
+		expect( store.dispatch ).toBeCalledWith( expectedAction, expectedPayload );
+	} );
+
+	it( 'validates legacy bank data correctly and sets it in the store on success', () => {
+		const wrapper = mount( BankData, {
+			localVue,
+			propsData: {
+				validateBankDataUrl: '/check-iban',
+				validateLegacyBankDataUrl: '/generate-iban',
+			},
+			store: createStore(),
+			mocks: {
+				$t: () => {},
+			},
+		} );
+		const store = wrapper.vm.$store;
+		store.dispatch = jest.fn();
+
+		const iban = wrapper.find( '#iban' );
+		wrapper.setData( { accountId: '34560517' } );
+		iban.trigger( 'blur' );
+		const bic = wrapper.find( '#bic' );
+		wrapper.setData( { bankId: '50010517' } );
+		bic.trigger( 'blur' );
+		const expectedAction = action( NS_BANKDATA, setBankData );
+		const expectedPayload = {
+			validationUrl: '/generate-iban',
+			requestParams: { accountNumber: '34560517', bankCode: '50010517' },
+		} as BankAccountRequest;
+
+		expect( store.dispatch ).toHaveBeenLastCalledWith( expectedAction, expectedPayload );
+	} );
+
+	it( 'disables BIC field for German IBANs and not for other values', () => {
+		const wrapper = mount( BankData, {
+			localVue,
+			store: createStore(),
+			mocks: {
+				$t: () => {},
+			},
+		} );
+
+		const iban = wrapper.find( '#iban' );
+		const bic = wrapper.find( '#bic' );
+
+		wrapper.setData( { accountId: 'DE12345605171238489890' } );
+		iban.trigger( 'blur' );
+		expect( bic.element.getAttribute( 'disabled' ) ).toMatch( 'disabled' );
+
+		wrapper.setData( { accountId: 'AT12345605171238489890' } );
+		iban.trigger( 'blur' );
+		expect( bic.element.getAttribute( 'disabled' ) ).toBeNull();
+
+		wrapper.setData( { accountId: '34560517' } );
+		iban.trigger( 'blur' );
+		expect( bic.element.getAttribute( 'disabled' ) ).toBeNull();
+	} );
+
+	it( 'renders changes from the store in the input fields', () => {
+		const wrapper = mount( BankData, {
+			localVue,
+			store: createStore(),
+			mocks: {
+				$t: () => {},
+			},
+		} );
+
+		const iban = wrapper.find( '#iban' );
+		const bic = wrapper.find( '#bic' );
+		wrapper.setData( { accountId: 'AT12345605171238489890', bankId: 'ABCDDEFFXXX' } );
+		expect( ( ( <HTMLInputElement> iban.element ).value ) ).toMatch( 'AT12345605171238489890' );
+		expect( ( ( <HTMLInputElement> bic.element ).value ) ).toMatch( 'ABCDDEFFXXX' );
```

Comment:
Is this because TypeScript complains? :thinking:

Reply:
Of course, I'm not a typecasting masochist :hocho::face_with_head_bandage:!

I have tried a ton of things but it's simply impossible to get the `value` without first casting to `HTMLInputElement`.